File: node/m101/hw1-3.js
var mongodb = require('mongodb');
var express = require('express');

var db = mongodb.Db('m101', new mongodb.Server('localhost', 27017), {safe: true});


var app = express.createServer()
app.use(express.methodOverride());

app.get('/hw1/:n', function(req, res) {
	db.open(function(err, db) {
		var collection = new mongodb.Collection(db, 'funnynumbers');
		collection.find({}, {limit: 1, skip: req.params.n, sort: 'value'}).toArray(function(err, results) {
			if (err) {
				console.log(err);
			}
			db.close();
			res.send(results[0].value.toString());
		});
	});
});

console.log('App started on port 8080');
app.listen(8080);

File: scala/m101/hw13.scala

import org.scalatra._
import com.mongodb.casbah.Imports._

class HelloWorldApp extends ScalatraFilter {
  get("/hw1/:skip") {
  	val skip = params("skip").toInt 

  	val record = Mongo("m101", "funnynumbers").find()
  		.skip(skip)
  		.sort(MongoDBObject("value" -> true))
  		.next
    record.getAs[Double]("value").get
  }
}

object Mongo {
	lazy val mongo = MongoConnection()
	def apply(db: String) = mongo(db)
	def apply(db: String, collection: String) = mongo(db)(collection)
	def close = mongo.close
}
